### PR TITLE
Fix issue#494

### LINKED ISSUE #494: [skycoin.gui:ERROR] Get balance failed: get unconfirmed spending failed when checking addresses balance
When I check-out the latest version from dev I get the following error on /balance?addrs=xxx requests:

[skycoin.gui:ERROR] Get balance failed: get unconfirmed spending failed when checking addresses balance: get unconfirmed spend error:unconfirmed transaction's IN: [SOMEREFERENCE] is not in unspent pool

Address: http://127.0.0.1:6420
Git commit: 45f48541244bd8b2e1e1ccdf743f935db8669a47
OS: Windows 10

### PR DESCRIPTION
**Pull request by iketheadore**
Fix #494 clean invalid unconfirmed transactions when start

### CODE REVIEW

File: src/visor/visor.go [hunk 1]
```diff
@@ -308,12 +338,21 @@ func (vs *Visor) ExecuteSignedBlock(b coin.SignedBlock) error {
 		return err
 	}
 
+	// the transactions in body might be changed in the following ExecuteBlock,
```
Comment (gz-c) [inline src/visor/visor.go]: ExecuteBlock should not be modifying the block pointer

**Comment by iketheadore** [comment]
Updated, check please

File: src/visor/blockchain.go [hunk 2]
```diff
@@ -151,23 +151,27 @@ func (bc *Blockchain) headSeq() int64 {
...
-func (bc *Blockchain) processBlock(b *coin.Block) error {
-	if !bc.isGenesisBlock(*b) {
-		if err := bc.verifyBlockHeader(*b); err != nil {
-			return err
+func (bc *Blockchain) processBlock(b coin.Block) (coin.Block, error) {
+	if !bc.isGenesisBlock(b) {
+		if err := bc.verifyBlockHeader(b); err != nil {
+			return coin.Block{}, err
 		}
 		txns, err := bc.processTransactions(b.Body.Transactions)
```
Comment (gz-c) [inline src/visor/blockchain.go]: `processTransactions` doesn't always return a copy of `txns`, needs to always return a copy

Comment (iketheadore) [reply src/visor/blockchain.go]: oh, yes, it's a slice.

**Pull request merged**